charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml

```diff
@@ -36,6 +36,110 @@ groups:
   - record: seed:hvpa_status_applied_vpa_recommendation_target:sum_by_container
     expr: sum(hvpa_status_applied_vpa_recommendation{recommendation="target"}) by (container, resource)
 
+  # Recording rules for the count of containers which have CPU usage at or above 80% of their request, aggregated by containers.
+  - record: seed:containers_over_cpu_scaling_threshold_total:count_by_container
+    expr: count(max(rate(container_cpu_usage_seconds_total[5m])) by (container, namespace) / max(kube_pod_container_resource_requests_cpu_cores) by (container, namespace) >= 0.8) by (container)
+
+  # Recording rules for the count of containers which have memory usage at or above 120% of their request, aggregated by containers.
+  - record: seed:containers_over_memory_scaling_threshold_total:count_by_container
+    expr: count(max(container_memory_working_set_bytes) by (container, namespace) / max(kube_pod_container_resource_requests_memory_bytes) by (container, namespace) >= 1.2) by (container)
+
+  # Recording rules for the count of containers which have CPU usage at or above 80% of their limits, aggregated by containers.
+  - record: seed:containers_over_cpu_limiting_threshold_total:count_by_container
+    expr: count(max(rate(container_cpu_usage_seconds_total[5m])) by (container, namespace) / max(kube_pod_container_resource_limits_cpu_cores) by (container, namespace) >= 0.8) by (container)
+
+  # Recording rules for the count of containers which have memory usage at or above 80% of their limits, aggregated by containers.
+  - record: seed:containers_over_memory_limiting_threshold_total:count_by_container
+    expr: count(max(container_memory_working_set_bytes) by (container, namespace) / max(kube_pod_container_resource_limits_memory_bytes) by (container, namespace) >= 0.8) by (container)
```

Comment:
The rationale behind using `max` is to be conservative. The idea is that if _any_ of the replicas of the container in the namespace is at or above the threshold of limits then we count it. Not just if the total of usage is above the total limits. This is especially important because we have seen that Kubernetes `Services` do a poor job of load balancing across replicas.

Reply:
Ok, it will still work just making sure that you were aware of this. Then `max` is good 👍